ccsm/baseline-ccsm-v1.0-psd01.md

```diff
@@ -216,25 +216,37 @@ A CCSM utilized in a BPI MUST support cryptographic algorithms based on commonly
 
+[[R2]](#r2) testability: libp2p as a very commonly used P2P message protocol supports end-to-end encryption, see [here](https://github.com/libp2p/specs/blob/master/noise/README.md) for the NOISE specification and relevant tests [here](https://github.com/libp2p/go-libp2p/blob/master/test-plans/cmd/ping/main.go). This shows that the requirement is testable. 
+
 #### **[R3]**	
 The CCSM utilized in a BPI MUST support CCSM Node Key Management incl. backup and recovery that adheres to established industry security standards such as the US Federal Information Processing Standard [[FIPS](#fips)] or ISO 27001 [[ISO27001](#iso27001)].
 
+[[R3]](#r3) testability: Given that there exist certification programs for [[FIPS](#fips)] or ISO 27001 [[ISO27001](#iso27001)], see for example the [NIST program]([[FIPS](#fips)] or ISO 27001 [[ISO27001](#iso27001)]), the requirement is testable.
+
 #### **[R4]**	
 The CCSM utilized in a BPI SHOULD support programmatic economic security assurances.
 
 *Note, economic security assurances such as those used in Proof-of-Stake consensus algorithms are designed to provide additional security assurances beyond those of cryptography in distributed systems. The security assurances are based on a system of economic incentives and disincentives for distributed system participants with the expressed goal that honest behavior of distributed system participants which enhances system security is in their economic self-interest. This is akin to determining if a cryptographic algorithm is secure or not, and what the level of security of said algorithm is, the security of a system of economic incentives and disincentives must be proven through a game theoretic security analysis.*
 
+[[R4]](#r4) testability: Programmatic economic security assurances are typically implemented within the CCSM's consensus protocol. Examples are Proof-of-Work for Bitcoin, and Proof-of-Stake for Ethereum, as common CCSMs. Ethereum's Proof-of-Stake protocol is specified [here](https://arxiv.org/abs/1710.09437) and tests of a security audited reference implementation can be found [here](https://github.com/prysmaticlabs/prysm/tree/develop/testing). Therefore, the requirement is testable.
+
 #### **[D1]**	
 The CCSM utilized in a BPI SHOULD be compatible with CCSM protocol execution in Trusted Execution Environments (TEE).
 
 *Note, a TEE is a secure area of a main processor. It guarantees code and data loaded inside to be protected with respect to confidentiality and integrity. A TEE as an isolated execution environment provides security features such as isolated execution, integrity of applications executing within the TEE, along with confidentiality of their assets.*
 
+[[D1]](#d1) testability: An example of a CCSM utilizing a TEE is Hyperledger Sawtooth with its [test suite](https://github.com/hyperledger/sawtooth-core/tree/main/cli/tests). Therefore, the requirement is testable.
```

Comment:
So is the logic that is a single example of something accomplishing the requirement exists and is testable, then the requirement itself should be able to be built out in such a way that is also testable?

Reply:
see my 1st comment